### PR TITLE
feat(terraform): add expanded actions to Terraform AWS IAM resources

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- added a new `metadata` field `extensions` to support features, which modifies on how the graph checks work
- added the extension `IAM_ACTION_EXPANSION` to expand wildcard IAM actions in AWS resources

Now it is possible to check for precise AWS IAM actions, even when the resource is using wildcards, ex.

check
```yaml
metadata:
  name: "Ensure CreateRole and CreatePolicy are not used"
  id: "CKV2_CUSTOM_1"
  category: "IAM"
  extensions:
    - IAM_ACTION_EXPANSION
definition:
  or:
    - cond_type: "attribute"
      resource_types:
        - "data.aws_iam_policy_document"
      attribute: "statement[?(@.effect == Allow)].actions[*]"
      operator: "jsonpath_not_within"
      value:
       - "iam:CreatePolicy"
       - "iam:CreateRole"
```

resource
```hcl
data "aws_iam_policy_document" "example" {
  statement {
    sid    = "Example"
    effect = "Allow"

    actions = [
      "iam:*",
      "ec2:*",
    ]
  }
}
```

result
```
Check: CKV2_CUSTOM_1: "Ensure something important"
	FAILED for resource: aws_iam_policy_document.example
	File: /main.tf:199-209

		199 | data "aws_iam_policy_document" "example" {
		200 |   statement {
		201 |     sid    = "Example"
		202 |     effect = "Allow"
		203 | 
		204 |     actions = [
		205 |       "iam:*",
		206 |       "ec2:*",
		207 |     ]
		208 |   }
		209 | }
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/3.Custom Policies/Examples.md [hunk 1]
```diff
@@ -479,3 +479,26 @@ definition:
     - prod
     - prod-eu
 ```
+
+## Using a jsonpath operator with IAM_ACTION_EXPANSION extension
+
+The following policy will fail, when `iam:CreatePolicy` or `iam:CreateRole` is used in `aws_iam_policy_document` data block by exact name or via wildcards.
```
Comment (tsmithv11) [inline docs/3.Custom Policies/Examples.md:485]:
```suggestion
The following policy will fail if iam:CreatePolicy or iam:CreateRole are used directly or through wildcards (e.g., iam:* or iam:Create*) in the aws_iam_policy_document data block.
```

File: docs/3.Custom Policies/Examples.md [hunk 2]
```diff
@@ -479,3 +479,26 @@ definition:
...
+
+```yaml
+metadata:
+  name: "Ensure CreateRole and CreatePolicy are not used"
+  id: "CKV2_CUSTOM_1"
+  category: "IAM"
+  extensions:
+    - IAM_ACTION_EXPANSION
```
Comment (tsmithv11) [inline docs/3.Custom Policies/Examples.md:493]:
```suggestion
  extensions: # Optional as IAM_ACTION_EXPANSION is enabled by default
    - IAM_ACTION_EXPANSION 
```

File: docs/3.Custom Policies/YAML Custom Policies.md [hunk 3]
```diff
@@ -45,6 +46,70 @@ metadata:
   guideline: "https://docs.bridgecrew.io/docs/ckv2_custom_1"
 ```
 
+### Extensions
+
+Extensions offer a way to enrich the data of the original resource to support use cases, which are not possible with given data.
+
+Available extensions:
+
+* IAM_ACTION_EXPANSION (enabled by default)
```
Comment (tsmithv11) [inline docs/3.Custom Policies/YAML Custom Policies.md:55]: WDYT about making it the inverse logic for this flag? IAM Action Expansion is always on unless you use extension: DISABLE_IAM_ACTION_EXPANSION?

File: docs/3.Custom Policies/YAML Custom Policies.md [hunk 4]
```diff
@@ -45,6 +46,70 @@ metadata:
...
+metadata:
+  id: "CKV2_CUSTOM_1"
+  ...
+  extensions: []
+```
+
+#### IAM_ACTION_EXPANSION extension
+
+By enabling the IAM_ACTION_EXPANSION extension additional data is added to the `Action` field of AWS Terraform resources.
+Every action with a `*` wildcard will be expanded to the actual actions to give the possibility to query easily for precise permission names.
```
Comment (tsmithv11) [inline docs/3.Custom Policies/YAML Custom Policies.md:79]:
```suggestion
By enabling the IAM_ACTION_EXPANSION extension, additional data is added to the `Action` field of AWS Terraform resources. Every action with a `*` wildcard will be expanded to the actual actions, making it easier to query for specific permission names.
```

**Comment by pazbechor** [comment]
@gruebel 
Closing this one as conflicts exists & inactivity :) 
Open it in the future if it's still relevant!

**Pull request closed without merging**